vm/chef/cookbooks/mattermost/files/mattermost-utils

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash -eu
+#
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#######################################
+# Creates a user for Mattermost application in MYSQL database.
+# Arguments:
+#   MySQL Hostname.
+#   MySQL Database Name.
+#   MySQL Username to be created.
+#   MySQL Password for username created.
+#######################################
+function create_db_user() {
+  local db_host="$1"
+  local db_name="$2"
+  local db_user="$3"
+  local db_pwd="$4"
+
+  echo >&2 "Creating database user..."
+  mysql -u root -e "CREATE USER '${db_user}'@'${db_host}' IDENTIFIED BY '${db_pwd}';"
+  mysql -u root -e "GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, INDEX, ALTER, CREATE TEMPORARY TABLES ON ${db_name}.* TO '${db_user}'@'${db_host}' IDENTIFIED BY '${db_pwd}';";
+  mysql -u root -e "FLUSH PRIVILEGES"
+}
+
+#######################################
+# Updates Mattermost configuration
+# Arguments:
+#   Data Source Name for connection between Mattermost and MySQL
+#   Mattermost Website Name
+#######################################
+function update_config() {
+  local -r dsn="$1"
+  local -r site_name="$2"
+  local -r driver="mysql"
+  local config_path=/opt/mattermost/config/config.json
+
+  echo >&2 "Updating Mattermost configuration..."
+  local config_contents="$(cat "${config_path}" \
```

Comment:
Consider

```
jq -r --arg driver "$driver" --arg dsn "$dsn" --arg site_name "$site_name"  '.TeamSettings.SiteName = $site_name 
    | .SqlSettings.DriverName = $driver 
    | .SqlSettings.DataSource = $dsn' "${config_path}"
```

The current code can fail in cases where one of the variables has a special character like double quote. (Though may be unlikely)

Reply:
Done.